r-package/grf/tests/testthat/test_forest_summaries.R

```diff
@@ -175,4 +178,21 @@ test_that("best linear projection is reasonable", {
 
   expect_lt(mean(abs(std.errs)), 2.5)
   expect_gt(mean(abs(std.errs)), 0.25)
+
+  blp.shifted <- best_linear_projection(forest.shifted.W, X[,1:2])
+  expect_equal(blp.all[, "Estimate"], blp.shifted[, "Estimate"], tol = 0.05)
+  expect_equal(blp.all[, "Std. Error"], blp.shifted[, "Std. Error"], tol = 0.05)
+
+  blp.2W <- best_linear_projection(forest.2W, X[,1:2])
+  expect_equal(blp.all[, "Estimate"]/2, blp.2W[, "Estimate"], tol = 0.05)
+})
+
+test_that("best linear projection works with a single covariate", {
+  n <- 200
+  p <- 5
+  X <- matrix(rnorm(n * p), n, p)
+  W <- rbinom(n, 1, 0.25 + 0.5 * (X[, 1] > 0))
+  Y <- pmax(X[, 1], 0) * W + X[, 2] + pmin(X[, 3], 0) + rnorm(n)
+  forest <- causal_forest(X, Y, W, num.trees = 50)
+  best_linear_projection(forest, X[, 1])
 })
```

Comment:
(extreme nitpick) add an empty line at the end of the file

Reply:
(there is an empty line, but github doesn't show it I think)